internals/aws/service_creator.go

```diff
@@ -114,20 +122,17 @@ func GetEncryptRequest(plaintext string, keyID string, kmsSvc kmsiface.KMSAPI) (
 }
 
 // parseRole parses a given role and performs a best effort attempt
-// to find and return the corresponding full ARN. The given role can
-// be either a (partial) ARN or the role name and can be prefixed with
-// "role/". Note that this is a best effort attempt so the returned value
-// is not guaranteed to always be a valid ARN.
-func parseRole(role string, stsSvc stsiface.STSAPI) (string, error) {
+// to find and return the corresponding full ARN. The given role can be:
+// - A role name (e.g. my-role)
+// - A role name, prefixed with "role/" (e.g. role/my-role)
+// - A full ARN (e.g. arn:aws:iam::123456789012:role/my-role)
+//
+// Note that this is a best effort attempt so the returned value is not guaranteed to
+// always be a valid ARN.
+func parseRole(role string, accountID string) string {
 	if strings.Contains(role, ":") {
```

Comment:
This is fairly fuzzy matching logic to check if the given string is already a valid role ARN, correct? Do we want to use something a bit more sophisticated here or not?

Reply:
This hasn't been changed.

As also documented in the comment, the function does only return valid output on valid input. This is oke for its use-case, as the given metadata is only stored when creating the service succeeds, which only happens when the role and key are valid.